src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Logging/HttpLoggingServiceCollectionExtensions.cs

```diff
@@ -1,72 +1,110 @@
+            o.LoggingFields |= HttpLoggingFields.Duration;
+        });
+
+        // Internal stuff for route processing:
+        _ = services.AddHttpRouteProcessor();
+        _ = services.AddHttpRouteUtilities();
+        return services;
     }
 
     /// <summary>
-    /// Adds components for incoming HTTP requests logging into <see cref="IServiceCollection"/>.
+    /// Enables enrichment redaction of HTTP logging.
     /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
-    /// <param name="section">The configuration section to bind <see cref="LoggingOptions"/> to.</param>
-    /// <returns>The value of <paramref name="services"/>.</returns>
-    /// <exception cref="ArgumentNullException">
-    /// Either <paramref name="services"/> or <paramref name="section"/> is <see langword="null" />.
-    /// </exception>
-    [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, typeof(LoggingOptions))]
-    [UnconditionalSuppressMessage(
-        "Trimming",
-        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
-        Justification = "Addressed with [DynamicDependency]")]
-    public static IServiceCollection AddHttpLogging(this IServiceCollection services, IConfigurationSection section)
+    /// <param name="services">The service collection.</param>
+    /// <param name="section">The configuration section with the redaction settings.</param>
+    /// <returns>The original service collection.</returns>
+    public static IServiceCollection AddHttpLoggingRedaction(this IServiceCollection services, IConfigurationSection section)
     {
-        _ = Throw.IfNull(services);
         _ = Throw.IfNull(section);
 
-        return AddHttpLoggingInternal(services, builder => builder.Bind(section));
+        return services.AddHttpLoggingRedaction(configureRedaction: o =>
+        {
+            o.RequestPathLoggingMode = section.GetSection(nameof(LoggingRedactionOptions.RequestPathLoggingMode)).Get<IncomingPathLoggingMode>();
+            o.RequestPathParameterRedactionMode = section.GetSection(nameof(LoggingRedactionOptions.RequestPathParameterRedactionMode)).Get<HttpRouteParameterRedactionMode>();
+            var paths = section.GetSection(nameof(LoggingRedactionOptions.ExcludePathStartsWith)).Get<string[]>();
+            if (paths != null)
+            {
+                foreach (var path in paths)
+                {
+                    _ = o.ExcludePathStartsWith.Add(path);
+                }
+            }
+
+            var routeParams = section.GetSection(nameof(LoggingRedactionOptions.RouteParameterDataClasses));
+            foreach (var entry in routeParams.GetChildren())
+            {
+                var taxonomy = entry.GetValue<string>(nameof(DataClassification.TaxonomyName));
```

Comment:
I don't really have a better answer here, but I'm pretty sure this isn't the right approach. I don't want developer using magic numbers in their config files to represent data classes, that's just a bug farm.